.brigade/brigade.ts

```diff
@@ -13,25 +11,23 @@ class MakeTargetJob extends Job {
     this.primaryContainer.workingDirectory = localPath
     this.primaryContainer.environment = env || {}
     this.primaryContainer.environment["SKIP_DOCKER"] = "true"
-    if (event.worker?.git?.ref) {
-      const matchStr = event.worker.git.ref.match(releaseTagRegex)
-      if (matchStr) {
-        this.primaryContainer.environment["VERSION"] = Array.from(matchStr)[1] as string
-      }
-    }
     this.primaryContainer.command = [ "make" ]
     this.primaryContainer.arguments = [ target ]
   }
 }
 
 // PushImageJob is a specialized job type for publishing Docker images.
 class PushImageJob extends MakeTargetJob {
-  constructor(target: string, event: Event) {
-    super(target, kanikoImg, event, {
+  constructor(target: string, event: Event, version?: string) {
+    const env = {
       "DOCKER_ORG": event.project.secrets.dockerhubOrg,
       "DOCKER_USERNAME": event.project.secrets.dockerhubUsername,
       "DOCKER_PASSWORD": event.project.secrets.dockerhubPassword
-    })
+    }
+    if (version) {
+      env["VERSION"] = version
```

Comment:
Just a curio here for me... `env` is declared as a `const` but here we mod it.  JS allows?

Reply:
Good question. `const` prevents another value from being assigned to `env`, but it does not prevent modification of the object that `env` points to. fwiw, TS pushes uses of `const` over `let` wherever possible and depending on compiler settings may even require it for variables whose value never changes.